lib/samson/hooks.rb

```diff
@@ -174,7 +174,7 @@ def with_callback(name, *hook_blocks)
       def only_callbacks_for_plugin(plugin_name, hook_name)
         original_hooks = @hooks[hook_name]
         @hooks[hook_name] = @hooks[hook_name].select do |proc|
-          proc.source_location.first.include?(plugin_name)
+          proc.source_location.first.include?("/#{plugin_name}/")
```

Comment:
explain?

Reply:
using exact matching since we had multiple plugins that included the letters "env"

Reply:
sorry, for some reason thought this was a regex not a string, so you're filtering by directory name...